worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -155,3 +155,23 @@ function fieldExists(document, fieldName) {
     } else
         return false;
 }
+
+function fieldHasStringValue(document, fieldName, value, includeSubDocuments) {
+
+    var fieldValues = document.getField(fieldName).getValues();
+
+    for each(var fieldValue in fieldValues){
+        if (fieldValue.isStringValue() && fieldValue.getStringValue() == value){
+            return true;
+        }
+    }
+
+    if (includeSubDocuments && document.hasSubdocuments()){
```

Comment:
- The `document.hasSubdocuments()` isn't necessary here.
- `.filter(...).findFirst().isPresent()` could just be `.anyMatch()`

        return includeSubDocuments && document.getSubdocuments().stream().anyMatch(
            function (x) {
                return fieldHasValue(x, fieldName, value, includeSubDocuments);
            });